File: lecture006/count01.java
import java.util.Scanner;
public class count01{

    public static void main(String[] args){
        Scanner scn=new Scanner (System.in);
        
        System.out.println("Enter number of elements");
        int n=scn.nextInt();
        
        System.out.println("The elements are: ");
        int[] arr=new int[n];
        for(int i=0;i<n;i++){
            arr[i]=scn.nextInt();  
        }
        zeroandone(arr);
        System.out.println("The swaped array are: ");
        int[] arr=new int[n];
        for(int i=0;i<n;i++){
            System.out.print(a[i]+" ");  
        }   
    }
    public static void zeroandone(int []arr)
    {
        int ptr=0;
        int itr=0;
        while(itr<arr.length)
        {
            if(arr[itr]==0){
                swap(arr[itr],arr[ptr]);
                ptr++;
        }
        itr++;
        }
    }
}

File: lecture006/count01.cpp
#include<iostream>
#include<vector>
using namespace std;

void zeroAndone(vector<int>& arr){
    int ptr=0;
    int itr=0;
    while(itr<arr.size())
    {
        if(arr[itr]==0){
            swap(arr[itr],arr[ptr]);
            ptr++;
        }
    itr++;  
    }
}
 int main(){
        int n;
        cout<<"enter the size of array : ";
        cin>>n;
        vector<int> arr(n,0);
        cout<<"enter array : ";
        for(int i=0;i<arr.size();i++){
            cin>>arr[i];
        }        
        zeroAndone(arr);
        cout<<"swaped array is : ";
        for(int i=0;i<arr.size() ;i++){
            cout<<arr[i]<<" ";
        }
}